Status/Debuffs/Debuff_Vulnerability.gd

```diff
@@ -3,6 +3,14 @@ class_name Debuff_Vulnerability
 ## Vulnerability buff increases the amount of damage an Entity takes from attacks
 
 # @Override
-func get_modified_stats(stats: EntityStats) -> EntityStats:
-	stats.damage_taken_increase += status_power
-	return stats
+func _init() -> void:
+	is_stat_modification = true
+	is_on_apply = true
+	modifier_name = StatDictBase.POSSIBLE_MODIFIER_NAMES.damage
+
+# @Override
+func init_status(in_caster: Entity, in_target: Entity) -> void:
+	status_caster = in_caster
+	status_target = in_target
+	targeted_modifier_dict = in_target.get_stat_component().get_stats().defense_modifier_dict
+
```

Comment:
Maybe instead of having the `is_stat_modification` and `is_on_apply` flags, we can override `on_apply` in any of these stat modifying statuses and call `_modify_stats_with_status` from the parent

Reply:
I suppose you would have to override `on_remove` as well, to make sure the stat change is removed

Reply:
I'm not sure to see right now, i'll check the code in more details later

Reply:
I think this is related to the above conversation with the `on_apply` and other types of activation for cards if I understood correctly ?

Reply:
Yes it's the same